### ISSUE TITLE
Uncaught TypeError: Cannot read property 'appendChild' of null

### ISSUE THREAD

**Issue by kevin-hiq** [kudago/smart-app-banner#9]
I get the error : Uncaught TypeError: Cannot read property 'appendChild' of null

Smartbanner.js line: 189 "document.body.appendChild(sb);"

But "sb" does get declared a few lines higher up "var sb = document.createElement('div');"

Any idea what might be the issue?

**Comment by CtrlF5** [comment]
just installed it, and this is the first error i have as well.

**Comment by glibin** [comment]
Could you provide a piece of code?

**Comment by CtrlF5** [comment]
found that this is due to the body not being initialised yet as code is placed in head. Might be worth putting a document.readyStateChange before the code is run.

As a test i replaced line: 189

```
            document.onreadystatechange = function () {
              if (document.readyState == "complete") {
                document.body.appendChild(sb);
              }
            }
```

which works as it waits for the body to be available before adding the element.

**Comment by glibin** [comment]
Other way is to put smartbanner initialization right after body close tag

**Comment by CtrlF5** [comment]
yes, i can place the code at the bottom, which i might choose to do so, but the example you have on install is then incorrect. so by taking your example the code fails without a catch for the the document to load.

**Comment by iamstarkov** [comment]
I faced this issue too

**Comment by iamstarkov** [comment]
right now README contains relevant example. prob this issue should be closed
